autoscheduler/frontend/src/redux/actions/courseCards.ts

```diff
@@ -212,3 +238,66 @@ export function updateCourseCard(index: number, courseCard: CourseCardOptions, t
     return dispatch(updateCourseCardSync(index, courseCard));
   };
 }
+
+export function clearCourseCards(): ClearCourseCardsAction {
+  return { type: CLEAR_COURSE_CARDS };
+}
+
+/**
+ * Helper function to fetch updated sections for a course card and keep selected sections
+ * @param courseCard course card to update
+ * @param term term to fetch sections for
+ */
+async function updateSectionsForCourseCard(
+  courseCard: SerializedCourseCardOptions,
+  term: string,
+): Promise<CourseCardOptions> {
+  return fetchCourseCardFrom(courseCard, term).then((newCourseCard) => {
+    // course is now updated with new sections, re-select sections from original courseCard
+
+
+    const selectedSections = new Set(courseCard.sections);
+
+    newCourseCard.sections.forEach((section) => {
+      // eslint-disable-next-line no-param-reassign
+      if (selectedSections.has(section.section.id)) section.selected = true;
+    });
+
+    return newCourseCard;
+  });
+}
+
+/**
+ * Creates an updated course card from courseCard by making an API call for updated
+ * section information
+ * @param courseCard Course card to update
+ * @param term Term to fetch section information from
+ */
+export function replaceCourseCards(
+  courseCards: SerializedCourseCardOptions[],
+  term: string,
+): ThunkAction<void, RootState, undefined, CourseCardAction> {
+  // data for sections might have changed since last visit, so create a new course card
+  // for each one asynchronously and update them with data from courseCards
+  return (dispatch): void => {
+    // if courseCards is improperly formatted or no cards are saved, create default
+    if (!Array.isArray(courseCards) || courseCards.length === 0) {
```

Comment:
What else would `courseCards` be besides an array? Is this different from null-checking?

Reply:
The default response is `{}` instead of `[]` since I didn't want to change the expected format, I could just change it to `[]` instead to make this easier to understand